golemfactory/yagna-service-erigon
# Upgrade to yagna v0.10.0 and yapapi v0.9.0

This PR resolves following JIRA tickets

 
- [APPS-537](https://golemproject.atlassian.net/browse/APPS-537)
- [APPS-603](https://golemproject.atlassian.net/browse/APPS-603)
- [APPS-605](https://golemproject.atlassian.net/browse/APPS-605)
- [APPS-606](https://golemproject.atlassian.net/browse/APPS-606)


## Description



### Why

We want to use the latest yagna & yapapi for compatibility.

### What

* Upgraded yagna to `v0.10.0`
* Upgraded yapapi to `v0.9.0`
* Removed old hacks and cleaned up code.
* Added an option to specify a different payment driver when running the service (`erc20` was having some issues when I was debugging this)


## Testing instructions

### Assumptions 

* Have a golem box running yagna on a dedicated subnet. (Yea, it's quite difficult because we don't have enough devices, so you are excused for not testing this one. I can present a demo of it working at a standup).

### How to test

Run the app on a dedicated subnet and check if it's working:
* `YAGNA_PAYMENT_DRIVER=zksync SUBNET_TAG=<MY_SUBNET> docker-compose up`
* Go to `localhost:8000` and try to get an erigon node.
